### PR TITLE
BPFD Api refactor (Core and kube) 

### LINKED ISSUE #232: Update static program to be on par with bpfctl load
static programs are loaded from a toml file. The syntax/data structures/code need to be on par with the bpfctl load command. In particular:
* Need to be able to load from URL (missing --from-image equivalent)
* Add support for tc and tracepoint program types (only xdp currently supported)
* There is a field `pub attach: Option<String>` that doesn't appear to be used.

### PR DESCRIPTION
**Pull request by astoycos**
fixes #232 and #177 And also goes a bit further. 

Some API and other cleanups: 

- [X] Remove old Rust based K8s API
- [X] Core -> Move the `Direction` Field Down into `AttachType` Since SingleAttach Programs have no need or context of direction. Also remove `from_image` and `path` and instead rely on an https://www.rfc-editor.org/rfc/rfc1738 conformant URL to specify bytecode location as either a local file or remote container image URL
- [x]  K8s API Alignment -> Align the new K8s CRD API to be more like the native core bpfd-api, specifically in regards to attach type (i.e add support for specifying TC direction and non networkmultiattach programs), Change all references to `Ebpf` to just `Bpf`
- [ ]  Update All Documentation around these API changes

### CODE REVIEW

**Comment by astoycos** [comment]
This is almost done, I need to breakup the image build workflow into two steps... One for pushing and one for building 

That way if folks make changes to the image-build.yaml workflow file the job can run on every PR to make sure we didn't break anything and then we can ONLY push those to our official repos if the PR is merged

File: bpfctl/src/main.rs [hunk 1]
```diff
@@ -32,18 +30,14 @@ enum Commands {
 
 #[derive(Args)]
 struct Load {
-    /// Optional: Extract bytecode from container, signals <PATH> is a
-    /// container image URL.
-    #[clap(long, conflicts_with("section_name"))]
-    from_image: bool,
-
     /// Required if "--from-image" is not present: Name of the ELF section from the object file.
```
Comment (astoycos) [inline bpfctl/src/main.rs:33]: nit:: Remove comment around required --from-image

File: bpfctl/src/main.rs [hunk 2]
```diff
@@ -32,18 +30,14 @@ enum Commands {
...
-    #[clap(short, long, default_value = "", required_unless_present("from_image"))]
+    #[clap(short, long, default_value = "")]
     section_name: String,
 
-    /// Required: Program to load.
+    /// Required: Location of Program Bytecode to load. Either Local file (file:///<path>) or bytecode
```
Comment (astoycos) [inline bpfctl/src/main.rs:37]: TODO add image for some possible client side validation here

**Comment by dave-tucker** [comment]
Protobuf API changes LGTM. I haven't looked at the code yet...

**Comment by astoycos** [comment]
@Billy99 This should be pretty much ready to go for your review :)

**Comment by astoycos** [comment]
++ @anfredette

**Comment by astoycos** [comment]
I'll work on adding integration tests but I think this PR is already large enough

File: bpfd-operator/api/v1alpha1/shared_types.go [hunk 3]
```diff
@@ -17,12 +17,63 @@ limitations under the License.
...
+// BpfProgramAttachPoint defines the allowed attach points
 // for a program loaded via bpfd Exactly one attach point must
 // be set.
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
-type EbpfProgramAttachPoint struct {
+type BpfProgramAttachPoint struct {
+	// NetworkMultiAttach defines an attach point for programs
+	// which attach to network devices and must be
+	// ordered via bpfd
```
Comment (astoycos) [inline bpfd-operator/api/v1alpha1/shared_types.go:28]: nit: period

File: bpfd-operator/pkg/bpfd-operator/bpfd-operator.go [hunk 4]
```diff
@@ -124,122 +125,142 @@ func (r *EbpfProgramConfigReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
 
 		return ctrl.Result{Requeue: false}, nil
 	}
 
-	failedEbpfPrograms := []string{}
-	// Make sure no ebpfPrograms have any issues issue
-	for _, ebpfProgram := range ebpfPrograms.Items {
+	failedBpfPrograms := []string{}
+	finalApplied := []string{}
+	// Make sure no bpfPrograms hade any issues in the loading process
```
Comment (astoycos) [inline bpfd-operator/pkg/bpfd-operator/bpfd-operator.go:214]: /s/hade/had

**Comment by astoycos** [comment]
Main commits are c435ebe25227df523f364bd2bea07f0b3fb19038, 26a708b78e8c803a9f9c35c9eea34f3a3fce0324, 1b0376f97bd7f10c03641de58b260e98fb6b382e

**Review by Billy99 — APPROVED** [review]
/lgtm

Couple of nits, but nothing worth holding the merge up on. Large PR and tried to time block the review. I'm going to merge and spend more time testing and walking through code this week.

File: bpfd-operator/api/v1alpha1/shared_types.go [hunk 5]
```diff
@@ -17,12 +17,63 @@ limitations under the License.
...
+	// SingleAttach defines an attach point for programs which
+	// attach to a single entity and do not need to be ordered.
+	// +optional
+	SingleAttach *BpfSingleAttach `json:"singleattach,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=ABORTED;DROP;PASS;TX;REDIRECT;DISPATCHER_RETURN
+type ProceedOnValue string
+
+// EbpfNetworkMultiAttach defines an ebpf attach
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/api/v1alpha1/shared_types.go:41]: Done

File: bpfd-operator/config/rbac/bpfd-agent-role/role.yaml [hunk 6]
```diff
@@ -8,15 +8,15 @@ rules:
 - apiGroups:
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/config/rbac/bpfd-agent-role/role.yaml:8]: Fixing

**Pull request merged**